scripts/bootstrap.sh

```diff
@@ -14,28 +14,77 @@
 # limitations under the License.
 #
 
-if [[ -n ${BASH_SOURCE[0]} ]]; then
-    CHIP_ROOT=$(cd "${BASH_SOURCE[0]%/*}/.." && pwd)
-else
-    CHIP_ROOT=$(cd "${0%/*}/.." && pwd)
-fi
+_bootstrap_or_activate() {
+    if [ -n "$BASH" ]; then
+        local _BOOTSTRAP_PATH="${BASH_SOURCE[0]}"
+    else
+        local _BOOTSTRAP_PATH="$1"
+    fi
 
-export PW_BRANDING_BANNER="$CHIP_ROOT/.chip-banner.txt"
-export PW_BRANDING_BANNER_COLOR="bold_white"
-export PW_VIRTUALENV_REQUIREMENTS="$CHIP_ROOT/scripts/requirements.txt"
+    local _BOOTSTRAP_NAME="${_BOOTSTRAP_PATH##*/}"
+    local _CHIP_ROOT="$(cd "${_BOOTSTRAP_PATH%/*}/.." && pwd)"
 
-if test -f "$CHIP_ROOT/scripts/helpers/rename_submodules.sh"; then
-    "$CHIP_ROOT/scripts/helpers/rename_submodules.sh"
-fi
+    if [ "$_BOOTSTRAP_NAME" = "bootstrap.sh" ]; then
+        git submodule update --init
```

Comment:
this is harmless to re-do anytime, right?

Reply:
> I have an issue with this, I want to run activate and have it bootstrap, just like pigweed does, if the tree hasn't been bootstrapped.
> 
> This appears to require that bootstrap be run once at some point. Do I misunderstand?
> I have an issue with this, I want to run activate and have it bootstrap, just like pigweed does, if the tree hasn't been bootstrapped.
> 
> This appears to require that bootstrap be run once at some point. Do I misunderstand?

That should work fine because the setup script won't exist yet, that's what the

`[ ! -f "$_SETUP_SH" ]`

part is for.

IMO this is still pretty flawed, as it doesn't really clue in that the environment need to be updated well, but it's the same code Pigweed was using so not a regression.

Reply:
> this is harmless to re-do anytime, right?

Unless you've made unstaged changes to those....

Maybe it's a niche enough case that we should just always do it (`gn_build.sh` also breaks this).

Reply:
I realized you're right, the submodule update needs to be done the first time for sure and I missed that. Updated.